src/Temporalio/Activities/ActivityCancellationDetails.cs

```diff
@@ -0,0 +1,42 @@
+namespace Temporalio.Activities
+{
+    /// <summary>
+    /// Details that are set when an activity is cancelled. This is only valid at the time the
+    /// cancel was received, the state may change on the server after it is received.
+    /// </summary>
+    public class ActivityCancellationDetails
+    {
+        /// <summary>
+        /// Gets a value indicating whether the activity no longer exists on the server (may already
+        /// be completed or its workflow may be completed).
+        /// </summary>
+        public bool IsGoneFromServer { get; init; }
+
+        /// <summary>
+        /// Gets a value indicating whether the activity was explicitly cancelled.
+        /// </summary>
+        public bool IsCancelRequested { get; init; }
+
+        /// <summary>
+        /// Gets a value indicating whether the activity timeout caused activity to be marked
+        /// cancelled.
```

Comment:
```suggestion
        /// Gets a value indicating whether the activity is timed out.
```

Technically this can be set true without it being the cause I think?

Reply:
Without it being the singular cause, true, but it is _one_ of the causes. I guess I was using the word "caused" here to clarify that this is not true if it is timed out, it is true if timeout was one of the causes that tripped cancellation. You can trip cancellation for something else and _then_ timeout.